contracts/test/ReentrancyChecker.sol

```diff
@@ -0,0 +1,143 @@
+pragma solidity 0.8.0;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import "@openzeppelin/contracts/token/ERC777/IERC777Recipient.sol";
+import "@openzeppelin/contracts/token/ERC777/IERC777Sender.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+
+contract ReentrancyChecker is
+    IERC1155Receiver,
+    IERC777Recipient,
+    IERC777Sender,
+    IERC721Receiver
+{
+    struct ExecParams {
+        address to;
+        bytes payload;
+        uint256 value;
+        uint256 count;
+        uint256 countSuccess;
+    }
+
+    ExecParams params;
+    uint256 maxRecursion = 2;
```

Comment:
Do we want to make max recursion a param so we can run it as many times as we want?

Reply:
Not sure. When you check for reentrancy just a single recursion is enough. Not sure if anyone would need to check for calls more than 1 level deep

Reply:
We can leave and if we find we need to do more than one recursion we can change later. I guess I was just trying to think of a situation where we would want to do more than one.